scipy/optimize/_minimize.py

```diff
@@ -597,53 +597,89 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
                       bounds=bounds, constraints=constraints,
                       callback=callback, **options)
 
+    if constraints is not None:
+        constraints = standardize_constraints(constraints, x0, meth)
+
+    remove_vars = False
     if bounds is not None:
         bounds = standardize_bounds(bounds, x0, meth)
 
-    if constraints is not None:
-        constraints = standardize_constraints(constraints, x0, meth)
+        lb, ub = _split_bounds(bounds)
+        btol = 1e-8  # tolerance for bounds being considered equal. Expose?
```

Comment:
I would set this to zero. It's not clear what the finite difference steps will be.